opm/simulators/wells/GuideRateHandler.cpp

```diff
@@ -631,33 +643,57 @@ GuideRateHandler<Scalar>::UpdateGuideRates::
 updateGuideRatesForWells_()
 {
     OPM_TIMEFUNCTION();
-    const auto &pu = this->phaseUsage();
-    for (const auto& well : this->schedule().getWells(this->report_step_idx_)) {
-        std::array<Scalar,3> potentials{};
-        auto& [oilpot, gaspot, waterpot] = potentials;
 
-        const auto& well_index = this->well_state_.index(well.name());
-        if (well_index.has_value() && this->well_state_.wellIsOwned(well_index.value(), well.name()))
-        {
-            // the well is found and owned
-            const auto& ws = this->well_state_.well(well_index.value());
-            const auto& wpot = ws.well_potentials;
-            if (pu.phase_used[BlackoilPhases::Liquid] > 0)
-                oilpot = wpot[pu.phase_pos[BlackoilPhases::Liquid]];
+    const auto& pu = this->phaseUsage();
+
+    const auto o_pos = (pu.phase_used[BlackoilPhases::Liquid] > 0)
+        ? pu.phase_pos[BlackoilPhases::Liquid] : -1;
 
-            if (pu.phase_used[BlackoilPhases::Vapour] > 0)
-                gaspot = wpot[pu.phase_pos[BlackoilPhases::Vapour]];
+    const auto g_pos = (pu.phase_used[BlackoilPhases::Vapour] > 0)
+        ? pu.phase_pos[BlackoilPhases::Vapour] : -1;
 
-            if (pu.phase_used[BlackoilPhases::Aqua] > 0)
-                waterpot = wpot[pu.phase_pos[BlackoilPhases::Aqua]];
+    const auto w_pos = (pu.phase_used[BlackoilPhases::Aqua] > 0)
+        ? pu.phase_pos[BlackoilPhases::Aqua] : -1;
+
+    const auto o_ix = std::size_t{0};
```

Comment:
these can be constexpr

Reply:
> these can be constexpr

Yep, you're right.  I've pushed an update to that effect now.